pkg/hub/client/public/iface_options.go

```diff
@@ -1,24 +1,41 @@
 package public
 
-// FindInterfaceRevisionOptions stores Interface Revision filtering parameters.
-type FindInterfaceRevisionOptions struct {
-	Fields string
-}
+import (
+	"fmt"
+
+	gqlpublicapi "capact.io/capact/pkg/hub/api/graphql/public"
+)
 
-// Apply is used to configure the ListImplementationRevisionsOptions.
-func (o *FindInterfaceRevisionOptions) Apply(opts ...FindInterfaceRevisionOption) {
-	o.Fields = InterfaceRevisionAllFields // default to all fields
+// InterfaceOption provides an option to configure the find request for Interface Revision.
+type InterfaceOption func(*InterfaceOptions)
+
+// InterfaceOptions stores Interface filtering parameters.
+type InterfaceOptions struct {
+	additionalFields string
+	filter           gqlpublicapi.InterfaceFilter
+}
 
+// Apply is used to configure the InterfaceOption.
+func (o *InterfaceOptions) Apply(opts ...InterfaceOption) {
 	// Apply overrides
 	for _, opt := range opts {
 		opt(o)
 	}
 }
 
-// FindInterfaceRevisionOption provides an option to configure the find request for Interface Revision.
-type FindInterfaceRevisionOption func(*FindInterfaceRevisionOptions)
+// WithLatestIfaceRevision adds latestRevision fields for Interface query.
+func WithLatestIfaceRevision(requestedFields InterfaceRevisionQueryFields) InterfaceOption {
```

Comment:
Honestly, I'm not a big fan of `Iface` in public func/method names 😄  I would suggest changing that to Interface, but it's up to you 🙂   Then we would need to rename the `WithImplRevCustomFields` option as well, which will be really, really long...

Reply:
ok, for public types/funcs I used the InterfaceRevision and ImplementationRevision form 👍 

I also removed the `custom` from the the `WithImplRevCustomFields` as it redundant.